pkg/actors/internal/reminders.go

```diff
@@ -31,12 +29,10 @@ type ExecuteReminderFn func(reminder *Reminder) bool
 // LookupActorFn is the type of a function that returns whether an actor is locally-hosted and the address of its host.
 type LookupActorFn func(ctx context.Context, actorType string, actorID string) (isLocal bool, actorAddress string)
 
-// StateStoreProviderFn is the type of a function that returns the state store provider.
-type StateStoreProviderFn func() (TransactionalStateStore, error)
+// StateStoreProviderFn is the type of a function that returns the state store provider and its name.
+type StateStoreProviderFn func() (string, TransactionalStateStore, error)
```

Comment:
till now, we were using state store name from ActorStateStoreName - is deriving this data from StateStoreProviderFn necessary?

Reply:
It allows not having to pass another option to `ActorsProviderOpts` since it is only needed by local providers. See: https://github.com/dapr/dapr/pull/7281#discussion_r1421879007